File: mailblastr-ruby/lib/mailblastr/polls.rb
# frozen_string_literal: true

module Mailblastr
  # Read-only results of the in-email poll widget.
  module Polls
    class << self
      # One summary row per email that has poll responses. GET /polls
      def list(params = {})
        Client.request(:get, "/polls", query: Client.pagination(params))
      end

      # The aggregated answer breakdown for one email. GET /polls/:email_id
      def get(email_id)
        Client.request(:get, "/polls/#{Client.path_escape(email_id)}")
      end
    end
  end
end

File: mailblastr-php/src/Resources/Polls.php
<?php

declare(strict_types=1);

namespace Mailblastr\Resources;

use Mailblastr\Client;

/**
 * Read-only results of the in-email poll widget.
 */
class Polls extends Resource
{
    /**
     * One summary row per email that has poll responses. GET /polls
     *
     * @param array $params Cursor pagination: limit, after, before.
     */
    public function list(array $params = []): array
    {
        return $this->client->request('GET', '/polls' . $this->paginationQuery($params));
    }

    /** The aggregated answer breakdown for one email. GET /polls/:emailId */
    public function get(string $emailId): array
    {
        return $this->client->request('GET', '/polls/' . Client::e($emailId));
    }
}
